File: benchmarks-QuTiP/coherentstate.py
import qutip as qt
import numpy as np
import benchmarkutils

name = "coherentstate"

samples = 5
evals = 100
cutoffs = range(50, 501, 50)

def setup(N):
    alpha = np.log(N)
    return alpha

def f(N, alpha):
    return qt.coherent(N, alpha, method="analytic")

print("Benchmarking:", name)
print("Cutoff: ", end="", flush=True)
checks = {}
results = []
for N in cutoffs:
    print(N, "", end="", flush=True)
    alpha = setup(N)
    checks[N] = qt.expect(qt.destroy(N), f(N, alpha))
    t = benchmarkutils.run_benchmark(f, N, alpha, samples=samples, evals=evals)
    results.append({"N": N, "t": t})
print()

benchmarkutils.check(name, checks, 0.05)
benchmarkutils.save(name, results)

File: benchmarks-QuantumOptics.jl/coherentstate.jl
using QuantumOptics
using BenchmarkTools
include("benchmarkutils.jl")

name = "coherentstate"

samples = 5
evals = 10000
cutoffs = [50:50:500;]

function setup(N)
    b = FockBasis(N-1)
    alpha = log(N)
    b, alpha
end

function f(b, alpha)
    coherentstate(b, alpha)
end

println("Benchmarking: ", name)
print("Cutoff: ")
checks = Dict{Int, Float64}()
results = []
for N in cutoffs
    print(N, " ")
    b, alpha = setup(N)
    checks[N] = abs(expect(destroy(b), f(b, alpha)))
    t = @belapsed f($b, $alpha) samples=samples evals=evals
    push!(results, Dict("N"=>N, "t"=>t))
end
println()

benchmarkutils.check(name, checks)
benchmarkutils.save(name, results)
